### ISSUE TITLE
1.14 - "cameraPosition" uniform is incorrect

### ISSUE THREAD

**Issue by zombye** [sp614x/optifine#2390]
`cameraPosition` is actually at the players feet, rather than at the actual camera.

This has actually always been the case as far as I know, but it hasn't really been an issue until 1.14.
Before 1.14, `gbufferModelView`/`gbufferModelViewInverse`/`gbufferPreviousModelView` would include an offset between the players feet and the actual location of the camera, which would counteract the effects of `cameraPosition` being at the players feet instead of at the camera.
In 1.14, this offset is no longer included in these matrices, which means that most shaders will be broken in some way.

**Comment by sp614x** [comment]
Wouldn't it be better if this is reversed?
- the `cameraPosition` should be at eye height
- the modelView should have no correction
A `shaders.properties` option could be added to switch modes.

**Comment by sp614x** [comment]
In 1.13 there is a difference between render and shadow modelView:
```
Projection shadow: 0.8027061, 0.0, 0.0, 0.0, 0.0, 1.428148, 0.0, 0.0, 0.0, 0.0, -1.0003138, -1.0, 0.0, 0.0, -0.100015685, 0.0
Projection render: 0.8027061, 0.0, 0.0, 0.0, 0.0, 1.428148, 0.0, 0.0, 0.0, 0.0, -1.0003138, -1.0, 0.0, 0.0, -0.100015685, 0.0
ModelView shadow: -1.0, 0.0, 8.742278E-8, 0.0, 0.0, 1.0, 0.0, 0.0, -8.742278E-8, 0.0, -1.0, 0.0, 0.0, -1.6199999, -4.0, 1.0
ModelView render: -1.0, 0.0, 8.742278E-8, 0.0, 0.0, 1.0, 0.0, 0.0, -8.742278E-8, 0.0, -1.0, 0.0, 0.0, -1.6199999, 0.05, 1.0
```

**Comment by sp614x** [comment]
In 1.13 there are also differences between first person, third person and third person front:
```
A. First person
 ModelView shadow: -1.0, 0.0, 8.742278E-8, 0.0, 0.0, 1.0, 0.0, 0.0, -8.742278E-8, 0.0, -1.0, 0.0, 0.0, -1.6199999, -4.0, 1.0
 ModelView render: -1.0, 0.0, 8.742278E-8, 0.0, 0.0, 1.0, 0.0, 0.0, -8.742278E-8, 0.0, -1.0, 0.0, 0.0, -1.6199999, 0.05, 1.0

B. Third person
 ModelView shadow: -1.0, 0.0, 8.742278E-8, 0.0, 0.0, 1.0, 0.0, 0.0, -8.742278E-8, 0.0, -1.0, 0.0, 0.0, -1.6199999, -4.0, 1.0
 ModelView render: -1.0, 0.0, 8.742278E-8, 0.0, 0.0, 1.0, 0.0, 0.0, -8.742278E-8, 0.0, -1.0, 0.0, 0.0, -1.6199999, -4.0, 1.0

C. Third person front
 ModelView shadow: -1.0, 0.0, 8.742278E-8, 0.0, 0.0, 1.0, 0.0, 0.0, -8.742278E-8, 0.0, -1.0, 0.0, 0.0, -1.6199999, -4.0, 1.0
 ModelView render: 1.0, 0.0, -1.7484555E-7, 0.0, 0.0, 1.0, 0.0, 0.0, 1.7484555E-7, 0.0, 1.0, 0.0, -3.496911E-7, -1.6199995, -4.0, 1.0
```

**Comment by zombye** [comment]
Yeah what I would want is for `cameraPosition` to actually be the position of the players camera, that's why the issue title is referring to `cameraPosition` and not the matrices.
I mentioned the matrices changing as that's the reason why `cameraPosition` not actually being the camera position is an issue now while it wasn't before.

But regardless of whether `cameraPosition` is or isn't the actual camera position, as long as `<gbuffer/shadow>ModelViewInverse[3].xyz + cameraPosition` is the position of the respective "camera", shaders will be able to calculate the correct position for stuff.

**Comment by sp614x** [comment]
In 1.14 the GL camera is positioned at the player eyes and `modelView` which comes from opengl doesn't have translate corrections. 
In fact the uniform `cameraPosition` is wrong, it is not updated to have the 1.14 translate corrections.

**Comment by sp614x** [comment]
Fixed in OptiFine 1.14_HD_U_F1_pre5
